### PR TITLE
Add Django 1.11 support and drop support for Django 1.8

### PR DESCRIPTION
**Pull request by khyang-udemy**
@arski @udemy/platform 

Fixes the broken `iterator()` issue in Django 1.11 with an PrefetchIterable.
Since the ModelIterable was introduced in Django 1.9, I decided to drop support for 1.8 instead of accommodating it.

Added support for 1.11

### CODE REVIEW

File: src/prefetch.py [hunk 1]
```diff
@@ -48,6 +48,15 @@ def __init__(self, **kwargs):
         super(PrefetchManager, self).__init__()
 
 
+class PrefetchIterable(query.ModelIterable):
+    def __iter__(self):
```
Comment (arski) [inline src/prefetch.py]: Wait does this imply we can only prefetch model instances? Does this even work in our code - we prefetch lots of aggregate values

Comment (khyang-udemy) [reply src/prefetch.py]: I'm not sure about the exact use case you're referring to but in Django 1.10, any references to the base QuerySet's iterator just calls the ModelIterable's __iter__: https://github.com/django/django/blob/1.10.8/django/db/models/query.py#L328

Plus all our tests pass 😄

**Comment by khyang-udemy** [comment]
The Python 2.7 tests are not happy about my Python 3 syntax. The other errors I'll look into later (probably tonight).

File: setup.cfg [hunk 2]
```diff
@@ -62,9 +62,9 @@ python_versions =
     3.5
 
 dependencies =
-    1.8: Django==1.8.19
     1.9: Django==1.9.13
     1.10: Django==1.10.8
+    1.11: Django=1.11.15
```
Comment (aristiden7o) [inline setup.cfg]: should be Django==1.11.15

Comment (khyang-udemy) [reply setup.cfg]: Nice catch!

File: src/prefetch.py (hunk 1 above)
Comment (arski) [reply src/prefetch.py]: Fair enough :)

**Comment by arski** [comment]
How did our tests pass with that :hmm:

File: src/prefetch.py [hunk 3]
```diff
@@ -68,6 +77,7 @@ def __init__(self, model=None, query=None, using=None,
         super(PrefetchQuerySet, self).__init__(model, query, using, **kwargs)
         self._prefetch = {}
         self.prefetch_definitions = prefetch_definitions
+        self._iterable_class = PrefetchIterable
```
Comment (arski) [inline src/prefetch.py]: Do we need to define it as a prop here? Cant we just call that class in def iterator?

Comment (khyang-udemy) [reply src/prefetch.py]: A line of code is worth a 1000 words:

https://github.com/django/django/blob/1.11.15/django/db/models/query.py#L1118

Also, if you compare _fetch_all to the 1.10.8 tag, you can see exactly why this structure broke.

**Comment by khyang-udemy** [comment]
I don't think our tests actually tried to save and pull from the database.

We do things like this:
https://github.com/udemy/django-prefetch/blob/master/tests/test_app/tests.py#L293-L295

...which wouldn't mind not have a db set.

File: src/prefetch.py (hunk 3 above)
Comment (arski) [reply src/prefetch.py]: Cool

**Comment by arski** [comment]
Ill try to reach out to Ionelto take over pip, until then will you create a pr to start using this? Thanks!

**Pull request merged**